src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang

```diff
@@ -48,7 +48,56 @@ module sonic-bgp-neighbor {
 
     container sonic-bgp-neighbor {
         container BGP_NEIGHBOR {
+            list BGP_NEIGHBOR_TEMPLATE_LIST {
+                description "This list is to support template based BGP neighbor configuration handled by bgpcfgd";
+                key "neighbor";
+
+                leaf neighbor {
+                    type inet:ip-address;
+                    description "BGP Neighbor address";
+                }
+
+                leaf asn {
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                    description "Peer AS number";
+                }
+
+                leaf holdtime {
+                    type uint16;
+                    description "Hold time";
+                }
+
+                leaf keepalive {
+                    type uint16;
+                    description "Keepalive interval";
+                }
+
+                leaf local_addr {
```

Comment:
Should we make sure that local address and Peer address belongs to same address family?

Reply:
I dont see any easy way to check this, do you have any easy check in mind?